ir/src/transition_constraints/graph.rs

```diff
@@ -84,6 +84,18 @@ impl AlgebraicGraph {
                 // add the expression.
                 Ok(self.insert_op(Operation::Add(lhs, rhs)))
             }
+            Expr::Exp(lhs, rhs) => {
+                // add base subexpression.
+                let lhs = self.insert_expr(*lhs, trace_columns)?;
+                // add exponent subexpression.
+                if let Expr::Constant(exponent) = *rhs {
+                    Ok(self.insert_op(Operation::Exp(lhs, exponent as usize)))
+                } else {
+                    Err(SemanticError::InvalidExprType(
+                        "Exponent can only be a constant".to_string(),
+                    ))
```

Comment:
I think this error would be more useful if it includes information about the `lhs` and `rhs`.